### PR TITLE
feat(validation): Consume specification change event

### PR DESCRIPTION
**Pull request by viacheslavkol**
### Purpose
Consume specification change event

### Approach
- Add kafka consumer
- Add byId method to Specifications client
- Update specification cache on SpecificationUpdatedEvent

### Changes Checklist
- [ ] **API Changes**: Document any API paths, methods, request or response bodies changed, added, or removed.
- [ ] **Database Schema Changes**: Indicate any database schema changes and their impact. Confirm that migration scripts were created.
- [ ] **Interface Version Changes**: Indicate any changes to interface versions.
- [ ] **Interface Dependencies**: Document added or removed dependencies.
- [ ] **Permissions**: Document any changes to permissions.
- [ ] **Logging**: Confirm that logging is appropriately handled.
- [x] **Unit Testing**: Confirm that changed classes were covered by unit tests.
- [x] **Integration Testing**: Confirm that changed logic was covered by integration tests.
- [x] **Manual Testing**: Confirm that changes were tested on local or dev environment.
- [x] **NEWS**: Confirm that the NEWS file is updated with relevant information about the changes made in this pull request.

### Related Issues
[MODQM-436](https://folio-org.atlassian.net/browse/MODQM-436)

### CODE REVIEW

File: src/main/java/org/folio/qm/messaging/deserializer/SpecificationUpdatedEventDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import java.util.Arrays;
+import lombok.RequiredArgsConstructor;
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.folio.rspec.domain.dto.SpecificationUpdatedEvent;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class SpecificationUpdatedEventDeserializer implements Deserializer<SpecificationUpdatedEvent> {
```
Comment (psmagin) [inline src/main/java/org/folio/qm/messaging/deserializer/SpecificationUpdatedEventDeserializer.java:14]: Why we can't use JsonDeserializer provided by libs?

File: src/main/java/org/folio/qm/service/impl/MarcSpecificationServiceImpl.java [hunk 2]
```diff
@@ -29,4 +34,19 @@ public SpecificationDto getSpecification(MarcFormat marcFormat) {
...
+  public void updateSpecificationCache(SpecificationUpdatedEvent specificationUpdate) {
+    var specification = specificationStorageClient.getSpecification(specificationUpdate.specificationId());
+    var cache = cacheManager.getCache(SPECIFICATION_STORAGE_CACHE);
+
+    if (cache == null) {
+      log.warn("updateSpecificationCache:: no cache found for " + SPECIFICATION_STORAGE_CACHE);
+      return;
+    }
+
+    cache.put(specificationUpdate.tenantId() + ":" + specification.getProfile().name(), specification);
```
Comment (psmagin) [inline src/main/java/org/folio/qm/service/impl/MarcSpecificationServiceImpl.java:48]: We have cache by MarcFormat and not by FamilyProfile

File: src/test/java/org/folio/qm/controller/SpecificationKafkaListenerIT.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+@IntegrationTest
+class SpecificationKafkaListenerIT extends BaseIT {
+  @Autowired
+  private KafkaAdmin kafkaAdmin;
+  @Autowired
+  private BeanFactory beanFactory;
+  @Autowired
+  private KafkaTopicsInitializer topicsInitializer;
+
+  @BeforeEach
```
Comment (psmagin) [inline src/test/java/org/folio/qm/controller/SpecificationKafkaListenerIT.java:38]: Probably should be once for the class

File: src/test/java/org/folio/qm/controller/SpecificationKafkaListenerIT.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    var configurableBeanFactory = (ConfigurableBeanFactory) beanFactory;
+    if (!configurableBeanFactory.containsBean(beanName)) {
+      configurableBeanFactory.registerSingleton(beanName, topic);
+      kafkaAdmin.initialize();
+      topicsInitializer.restartEventListeners();
+    }
+  }
+
+  @Test
+  void specificationUpdatedEvent() throws Exception {
```
Comment (psmagin) [inline src/test/java/org/folio/qm/controller/SpecificationKafkaListenerIT.java:51]: Not clear test flow, please add comments

File: src/main/java/org/folio/qm/service/impl/MarcSpecificationServiceImpl.java (hunk 2 above)
Comment (viacheslavkol) [reply src/main/java/org/folio/qm/service/impl/MarcSpecificationServiceImpl.java:48]: Yes, but they have same names

**Review by psmagin — APPROVED** [review]

**Review by mukhiddin-yusuf — APPROVED** [review]

**Review by viacheslavpoliakov — APPROVED** [review]

**Pull request merged**